ui/components/app/modals/transaction-already-confirmed/transaction-already-confirmed.container.tsx

```diff
@@ -0,0 +1,47 @@
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { getBlockExplorerLink } from '@metamask/etherscan-link';
+import { type TransactionMeta } from '@metamask/transaction-controller';
+import { type NetworkClientConfiguration } from '@metamask/network-controller';
+
+import {
+  MetaMaskReduxDispatch,
+  MetaMaskReduxState,
+} from '../../../../store/store';
+import withModalProps from '../../../../helpers/higher-order-components/with-modal-props';
+import { hideModal } from '../../../../store/actions';
+import { getRpcPrefsForCurrentProvider } from '../../../../selectors';
+import TransactionAlreadyConfirmed from './transaction-already-confirmed.component';
+
+function mapDispatchToProps(dispatch: MetaMaskReduxDispatch) {
```

Comment:
Is this not the legacy pattern for components where we map to the state rather than being purely functional and relying on hooks like `useSelector` to obtain all the Redux state?

I can see we're using `withModalProps` which may be pushing us, but there's not much in it so could we create an equivalent hook like `useModalProps` that provides the same state and dispatch functions?

It's a very small investment now but would mean future modal components are single functional components without the need for the attached container?

Reply:
I didn't want to make it since there were already different approaches but since you called it, here it is 
https://github.com/MetaMask/metamask-extension/pull/22943/commits/90a3c852b6ababde598044df27bb0fa7666da947